### PR TITLE
withServices POC and Contract types

### PR DESCRIPTION
**Pull request by sebruiz**

### CODE REVIEW

**Comment by thehannahlerman** [comment]
Beautiful, LGTM after first pass but I will do some testing

**Review by thehannahlerman — APPROVED** [review]

**Comment by thehannahlerman** [comment]
Not sure if this is due to Vercel environment stuff, but loading the /all-questions page on this branch gives me a constant loading state

**Comment by jdaly13** [comment]
@sebruiz - can you look in the deployment process for this?  https://xyz-ifyv3uwv4-metricsdao.vercel.app/all-questions - gives me a constant loading state but I'm sure its different locally

**Comment by jdaly13** [comment]
Can you also update the README as well for any new steps in getting the app to run locally?

**Comment by jdaly13** [comment]
I'm thinking if we generate contracts on every build we probably shouldn't commit them and put that folder in the .gitignore?

**Comment by jdaly13** [comment]
> Not sure if this is due to Vercel environment stuff, but loading the /all-questions page on this branch gives me a constant loading state

The env variables should be up to date and point to polygon and the appropriate IPFS - for example the other deployment branches should works as well - I'm thinking it has to do with the generate:contracts script

File: app/services/contracts/question-state-controller-service.server.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+export default class QuestionStateControllerService {
+  private questionStateController: ContractContext;
+
+  constructor(provider: ethers.providers.Provider) {
+    this.questionStateController = new ethers.Contract(
+      questionStateControllerJson.address,
+      questionStateControllerJson.abi,
+      provider
+    ) as unknown as ContractContext;
```
Comment (jdaly13) [inline app/services/contracts/question-state-controller-service.server.ts:16]: this slightly confuses me - is this needed? Or maybe put a TODO to update unknown - not a biggie.

File: app/services/question-service.server.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  constructor(
+    private bountyQuestionService: BountyQuestionService,
+    private questionStateControllerService: QuestionStateControllerService
+  ) {}
+
+  //TODO: use this
+  async getQuestions() {
+    const currentQuestion = await this.bountyQuestionService.getCurrentQuestion();
+    if (!currentQuestion) {
+      return [];
```
Comment (jdaly13) [inline app/services/question-service.server.ts:15]: HMMM I'm think we should fail or give an error message if we can't get current question on the all questions page without that we really can't get all questions

File: bin/get-contracts.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+const fs = require("fs");
+const dotenv = require("dotenv");
+const { exec } = require("child_process");
+dotenv.config();
+
+const CORE_EVM_CONTRACTS = [
```
Comment (jdaly13) [inline bin/get-contracts.js:6]: Hmm instead of hardcoding these I wonder if we could get them directly from the core-evm-package

**Comment by thehannahlerman** [comment]
> @sebruiz - can you look in the deployment process for this? https://xyz-ifyv3uwv4-metricsdao.vercel.app/all-questions - gives me a constant loading state but I'm sure its different locally

same for me

**Comment by sebruiz** [comment]
Need to rethink changing contracts with switchNetwork. Probably want to avoid server-side chain calls as much as possible.

**Pull request closed without merging**